gnovm/pkg/gnolang/helpers.go

```diff
@@ -10,39 +10,34 @@ import (
 // ----------------------------------------
 // Functions centralizing definitions
 
-// RealmPathPrefix is the prefix used to identify pkgpaths which are meant to
-// be realms and as such to have their state persisted. This is used by [IsRealmPath].
-const (
-	RealmPathPrefix   = "gno.land/r/"
-	PackagePathPrefix = "gno.land/p/"
+// ReRealmPath and RePackagePath are the regexes used to identify pkgpaths which are meant to
+// be realms with persisted states and pure packages.
+var (
+	ReRealmPath   = regexp.MustCompile(`^([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.[a-zA-Z]{2,}/r/[a-z0-9_/]+`)
+	RePackagePath = regexp.MustCompile(`^([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.[a-zA-Z]{2,}/p/[a-z0-9_/]+`)
 )
 
 // ReGnoRunPath is the path used for realms executed in maketx run.
-// These are not considered realms, as an exception to the RealmPathPrefix rule.
-var ReGnoRunPath = regexp.MustCompile(`^gno\.land/r/g[a-z0-9]+/run$`)
+// These are not considered realms, as an exception to the ReRealmPathPrefix rule.
+var ReGnoRunPath = regexp.MustCompile(`^([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.[a-zA-Z]{2,}/r/g[a-z0-9]+/run$`)
 
 // IsRealmPath determines whether the given pkgpath is for a realm, and as such
 // should persist the global state.
 func IsRealmPath(pkgPath string) bool {
-	return strings.HasPrefix(pkgPath, RealmPathPrefix) &&
-		// MsgRun pkgPath aren't realms
+	return ReRealmPath.MatchString(pkgPath) &&
 		!ReGnoRunPath.MatchString(pkgPath)
 }
 
 // IsPurePackagePath determines whether the given pkgpath is for a published Gno package.
 // It only considers "pure" those starting with gno.land/p/, so it returns false for
 // stdlib packages and MsgRun paths.
 func IsPurePackagePath(pkgPath string) bool {
-	return strings.HasPrefix(pkgPath, PackagePathPrefix)
+	return RePackagePath.MatchString(pkgPath)
 }
 
 // IsStdlib determines whether s is a pkgpath for a standard library.
 func IsStdlib(s string) bool {
-	// NOTE(morgan): this is likely to change in the future as we add support for
-	// IBC/ICS and we allow import paths to other chains. It might be good to
-	// (eventually) follow the same rule as Go, which is: does the first
-	// element of the import path contain a dot?
-	return !strings.HasPrefix(s, "gno.land/")
+	return strings.IndexByte(s[:strings.IndexByte(s, '/')+1], '.') < 0
```

Comment:
`"morgan.land"` would be a standard library, best to consider whole of `s` if IndexByte < 0

Reply:
```suggestion
	idx := strings.IndexByte(s, '/')
	if idx < 0 {
		// If no '/' is found, consider the whole string
		return strings.IndexByte(s, '.') < 0
	}
	return strings.IndexByte(s[:idx+1], '.') < 0
```